crates/c-api/src/types/instance.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{wasm_externtype_t, wasm_limits_t, CExternType};
+use once_cell::unsync::OnceCell;
+use wasmtime::InstanceType;
+
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct wasm_instancetype_t {
+    ext: wasm_externtype_t,
+}
+
+wasmtime_c_api_macros::declare_ty!(wasm_instancetype_t);
+
+#[derive(Clone)]
+pub(crate) struct CInstanceType {
+    pub(crate) ty: InstanceType,
+    limits_cache: OnceCell<wasm_limits_t>,
+}
+
+impl wasm_instancetype_t {
+    // pub(crate) fn new(ty: InstanceType) -> wasm_instancetype_t {
+    //     wasm_instancetype_t {
+    //         ext: wasm_externtype_t::new(ty.into()),
+    //     }
+    // }
+
+    pub(crate) fn try_from(e: &wasm_externtype_t) -> Option<&wasm_instancetype_t> {
+        match &e.which {
+            CExternType::Instance(_) => Some(unsafe { &*(e as *const _ as *const _) }),
+            _ => None,
+        }
+    }
+
+    // pub(crate) fn ty(&self) -> &CInstanceType {
+    //     match &self.ext.which {
+    //         CExternType::Instance(f) => &f,
+    //         _ => unsafe { std::hint::unreachable_unchecked() },
+    //     }
+    // }
```

Comment:
Commented out code?

Also, is getting the `ty` so perf sensitive that we have to use potentially-UB-inducing `unreachable_unchecked` over regular `unreachable`?

Reply:
I guess this is going into an FFI language boundary... Did the auto-abort on unwind across FFI stuff re-merge? If so, then we should use `unreachable!()`, if not then we should abort ourselves I think.

Reply:
Oh this was copied from the other types in the C API, but I think it makes more sense to just add this back in as necessary later on.

Also yeah I doubt that `unreachable_unchecked` is really worth it here, and an `unreachable!()` should be fine.